### PR TITLE
Update to use ArcGIS runtime u10

### PR DESCRIPTION
**Pull request by puneet-pdx**
- Update versions - 
   Android Gradle Plugin Version - 4.1.2
   Gradle version - 6.5
   Min SDK - 23 required by ArcGIS runtime
   ArcGIS runtime version - 100.10.0
   test runner version
   materialLibraryVersion
   appcompatLibraryVersion

- Update app versionName to 1.0.11

### CODE REVIEW

**Review by sorenoid — APPROVED** [review]
👍

**Review by esreli — CHANGES_REQUESTED** [review]
This PR looks good to me. Thanks for updating the project for U10 and certifying it still performs as expected.

Since the launch of the new dev-x site, I've noticed a number of broken links.

### README

- [license your app](https://github.com/Esri/maps-app-android/tree/puneet/updateToUseU10#licensing) leads to a `404`. Can we update the link to use the new dev-x site [link](https://developers.arcgis.com/android/license-and-deployment/)?

### docs/README

- [geocoding parameters](https://github.com/Esri/maps-app-android/tree/puneet/updateToUseU10/docs#geocoding) links to a `404`. 
- [The loadable pattern is described here.](https://github.com/Esri/maps-app-android/tree/puneet/updateToUseU10/docs#place-search) - links to `404`.
- [Portal API](https://github.com/Esri/maps-app-android/tree/puneet/updateToUseU10/docs#accessing-your-organizations-basemaps) - links to `404`.

Are there different links you can find to replace these?

**Comment by puneet-pdx** [comment]
@esreli I have addressed your comments. Please have another look.

File: README.md [hunk 1]
```diff
@@ -152,7 +152,7 @@ Unless required by applicable law or agreed to in writing, software distributed
 
 A copy of the license is available in the repository's [license.txt](https://github.com/Esri/maps-app-android/blob/master/license.txt) file.
 
-For information about licensing your deployed app, see [License your app](https://developers.arcgis.com/android/guide/license-your-app.htm).
+For information about licensing your deployed app, see [License your app](https://developers.arcgis.com/android/license-and-deployment).
```
Comment (esreli) [inline README.md:155]:
```suggestion
For information about licensing your deployed app, see [License your app](https://developers.arcgis.com/android/license-and-deployment/).
```

File: build.gradle [hunk 2]
```diff
@@ -20,21 +20,21 @@ allprojects {
...
+    materialLibraryVersion = '1.3.0'
     legacyLibraryVersion = '1.0.0'
 
-    runtimeVersion = '100.9.0'
+    runtimeVersion = '100.10.0'
     butterknifeVersion = '10.2.0'
     robotiumVersion = '5.6.3'
     httpcomponentsVersion = '4.3.5.1'
-    runnerVersion = '1.1.0'
+    runnerVersion = '1.3.0'
```
Comment (esreli) [inline build.gradle:37]: What is this?

File: docs/README.md [hunk 3]
```diff
@@ -35,7 +35,7 @@ mapView.setMap(map);
 
 ### Accessing your organization's basemaps
 
-As an administrator of an ArcGIS Online organization or Portal, you can configure the basemaps that your users can switch between via a [group](http://doc.arcgis.com/en/arcgis-online/share-maps/share-items.htm). Applications can leverage this configuration using the [Portal API](https://developers.arcgis.com/android/latest/guide/access-the-arcgis-platform.htm#ESRI_SECTION2_B8EDBBD3D4F1499C80AF43CFA73B8292). The Maps App does this by an asynchronous call to find the group containing web maps in the basemap gallery. With the returned group id, the collection of basemaps is retrieved from the portal.
+As an administrator of an ArcGIS Online organization or Portal, you can configure the basemaps that your users can switch between via a [group](http://doc.arcgis.com/en/arcgis-online/share-maps/share-items.htm). Applications can leverage this configuration using the [Portal API](https://developers.arcgis.com/android/arcgis-organization-portals). The Maps App does this by an asynchronous call to find the group containing web maps in the basemap gallery. With the returned group id, the collection of basemaps is retrieved from the portal.
```
Comment (esreli) [inline docs/README.md:38]:
```suggestion
As an administrator of an ArcGIS Online organization or Portal, you can configure the basemaps that your users can switch between via a [group](http://doc.arcgis.com/en/arcgis-online/share-maps/share-items.htm). Applications can leverage this configuration using the [Portal API](https://developers.arcgis.com/android/arcgis-organization-portals/). The Maps App does this by an asynchronous call to find the group containing web maps in the basemap gallery. With the returned group id, the collection of basemaps is retrieved from the portal.
```

File: docs/README.md [hunk 4]
```diff
@@ -129,7 +129,7 @@ In the Maps App, `LocatorTask`s are initialized using an online locator provided
...
+You can also provision your own [custom geocode service](https://doc.arcgis.com/en/arcgis-online/administer/configure-services.htm#ESRI_SECTION1_0A9A071A7AB748028C8213D1D863FA18) to support your organization. Before using the `LocatorTask` to geocode or search for places, the `LocatorTask` must be `LOADED`. The loadable pattern is described [here](https://developers.arcgis.com/android/programming-patterns/loadable). `LocatorTask` operations are performed asynchronously using `ListenableFutures`, an implementation of Java’s Future interface. `ListenableFutures` add the ability to attach a listener that runs upon completion of the task. One of the first user interactions the Maps App supports is suggesting places near the device location.
```
Comment (esreli) [inline docs/README.md:132]:
```suggestion
You can also provision your own [custom geocode service](https://doc.arcgis.com/en/arcgis-online/administer/configure-services.htm#ESRI_SECTION1_0A9A071A7AB748028C8213D1D863FA18) to support your organization. Before using the `LocatorTask` to geocode or search for places, the `LocatorTask` must be `LOADED`. The loadable pattern is described [here](https://developers.arcgis.com/android/programming-patterns/loadable/). `LocatorTask` operations are performed asynchronously using `ListenableFutures`, an implementation of Java’s Future interface. `ListenableFutures` add the ability to attach a listener that runs upon completion of the task. One of the first user interactions the Maps App supports is suggesting places near the device location.
```

**Review by esreli — APPROVED** [review]
I think this is ready to go 👍

File: build.gradle (hunk 2 above)
Comment (puneet-pdx) [reply build.gradle:37]: version for android test runner

**Pull request merged**